packages/destination-actions/src/destinations/salesforce/opportunity/index.ts

```diff
@@ -45,20 +48,34 @@ const action: ActionDefinition<Settings, Payload> = {
       if (!payload.close_date || !payload.name || !payload.stage_name) {
         throw new IntegrationError('Missing close_date, name or stage_name value', 'Misconfigured required field', 400)
       }
-      return await sf.createRecord(payload, 'Opportunity')
+      return await sf.createRecord(payload, OBJECT_NAME)
     }
 
     validateLookup(payload)
 
     if (payload.operation === 'update') {
-      return await sf.updateRecord(payload, 'Opportunity')
+      return await sf.updateRecord(payload, OBJECT_NAME)
     }
 
     if (payload.operation === 'upsert') {
       if (!payload.close_date || !payload.name || !payload.stage_name) {
         throw new IntegrationError('Missing close_date, name or stage_name value', 'Misconfigured required field', 400)
       }
-      return await sf.upsertRecord(payload, 'Opportunity')
+      return await sf.upsertRecord(payload, OBJECT_NAME)
+    }
+  },
+  performBatch: async (request, { settings, payload }) => {
+    const sf: Salesforce = new Salesforce(settings.instanceUrl, request)
+
+    if (payload[0].operation === 'bulkUpsert') {
+      if (!payload[0].close_date || !payload[0].name || !payload[0].stage_name) {
+        throw new IntegrationError('Missing close_date, name or stage_name value', 'Misconfigured required field', 400)
+      }
+
+      return await sf.bulkUpsert(payload, OBJECT_NAME)
+    } else {
+      const errorMsg = 'Bulk Upsert action must be used with batching'
+      throw new IntegrationError(errorMsg, errorMsg, 400)
```

Comment:
Not blocking but I guess the second argument should is an optional code.

```suggestion
      const errorMsg = 'Bulk Upsert action must be used with batching'
      throw new IntegrationError(errorMsg, 'Bad Request', 400)
```

https://github.com/segmentio/action-destinations/blob/4a53c8bbb3a07d4b4aa0b391d377ef1ccd58ca3f/packages/destination-actions/src/destinations/slack/postToChannel/index.ts#L50

Reply:
(As a larger follow-up we should make `IntegrationError` easier to get right. I see a lot of different usage of it across action-destinations)